src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py

```diff
@@ -159,20 +159,36 @@ def launch(
 
         settings = cast(SagemakerStepOperatorSettings, self.get_settings(info))
 
-        session = sagemaker.Session(default_bucket=self.config.bucket)
-        instance_type = settings.instance_type or "ml.m5.large"
+        # Get and default fill SageMaker estimator arguments 
+        estimator_args = settings.estimator_args
+        session = estimator_args.sagemaker_session or sagemaker.Session(default_bucket=self.config.bucket)
```

Comment:
All settings attributes need to be json serializable which I don't think is possible for the Sagemaker session, so that would not be something that's allowed

Reply:
In the `develop` branch, the Sagemaker session is initialized in the same way: https://github.com/zenml-io/zenml/blob/develop/src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py#L162

I'm thinking it would thus work, esp. since it's passed to a SageMaker Python module, which probably takes care of json serialization?

Above code would allow users to use a self-initialized SageMaker session and if not, a default session is initialized.

Looking forward to your thoughts!

Reply:
Yes, the problem won't be the initialization of the session, but rather the `estimator_args["sagemaker_session"]`. This dictionary comes from ZenML and can only contain json serializable values and therefore can't contain a valid sagemaker session.

Reply:
In https://github.com/zenml-io/zenml/pull/1381/commits/d95b2eb2adcd97691cd49775861fceb465eb4b74, the user provided `sagemaker_session` is overruled with a default Session. This does not avoid the user providing a value that cannot be json serialized, but at least discourages the user to provide custom sessions (and run into this problem). The current type of `estimator_args` in the settings class (`estimator_args: Dict[str, Any] = {}`) does not prevent the user from doing that.